app/models/Book.js

```diff
@@ -0,0 +1,14 @@
+class Book {
+  constructor(id, createdAt, updatedAt, title, author, publisher, yearOfPublication, numberOfPages) {
+    this.id = id;
+    this.createdAt = createdAt;
+    this.updatedAt = updatedAt;
+    this.title = title;
+    this.author = author;
+    this.publisher = publisher;
+    this.yearOfPublication = yearOfPublication;
+    this.numberOfPages = numberOfPages;
+  }
+
+  // Add validation methods and relationships here if necessary
+}
```

Comment:
Consider adding validation methods to ensure that the data being assigned to the Book class properties is valid. This will help maintain data integrity and prevent potential issues with invalid data being processed.